presto-hive/src/main/java/io/prestosql/plugin/hive/HiveQlToPrestoTranslator.java

```diff
@@ -0,0 +1,149 @@
+import com.google.common.collect.Lists;
+import com.google.common.collect.PeekingIterator;
+import io.prestosql.spi.PrestoException;
+
+import static io.prestosql.plugin.hive.HiveErrorCode.HIVE_TRANSLATION_ERROR;
+import static java.lang.String.format;
+
+/**
+ * Translate statements in Hive QL to Presto SQL.
+ *
+ * Only translation of quoted literals is currently included.
+ */
+public final class HiveQlToPrestoTranslator
+{
+    // Translation methods consume data from the iterator
+    private final PeekingIterator<Character> input;
+    private final StringBuilder output = new StringBuilder();
+
+    /**
+     * Translate a HiveQL statement to Presto SQL by fixing quoted identifiers
+     * and string literals. No other translation is included.
+     *
+     * <p>Backquotes are replaced with double quotes, including SQL-style
+     * escapes (`` becomes ` and " becomes "").
+     *
+     * <p>Single and double quotes are replaced with single quotes, with
+     * minimal processing of escape sequences to ensure that the strings end in
+     * the right place.
+     */
+    public static String translateHiveQuotesToPresto(String hiveStatement)
+    {
+        HiveQlToPrestoTranslator translator =
+                new HiveQlToPrestoTranslator(hiveStatement);
+        return translator.translateQuotedLiterals();
+    }
+
+    private HiveQlToPrestoTranslator(String hiveQl)
+    {
+        input = Iterators.peekingIterator(Lists.charactersOf(hiveQl).iterator());
+    }
+
+    private String translateQuotedLiterals()
+    {
+        // Consume input, passing control to other translation methods when
+        // their delimiters are encountered.
+        while (input.hasNext()) {
+            char c = input.next();
+            switch (c) {
+                case '"':
+                case '\'':
+                    translateString(c);
+                    break;
+                case '`':
+                    translateQuotedIdentifier();
+                    break;
+                default:
+                    output.append(c);
+                    break;
+            }
+        }
+
+        return output.toString();
+    }
+
+    private void translateString(char delimiter)
+    {
+        output.append("'");
+        while (input.hasNext()) {
+            char c = input.next();
+
+            // The possible outcomes in a string are (assuming "-quoting):
+            // - find delimiter and end the string
+            // - find backslash and treat following character as non-delimiter
+            // - replace ' with ''
+            // - leave any other character as-is
+            if (c == delimiter) {
+                output.append("'");
+                return;
+            }
+            else if (c == '\\') {
+                if (!input.hasNext()) {
+                    break; // skip to end-of-input error
+                }
+                // Don't handle escape sequences, just drop the backslash.
```

Comment:
This seems wrong. We should fail if there are escape sequences if we're not going to handle them. Otherwise, we are returning a known invalid result.

Reply:
That's a good idea. I originally tried to figure out what escape sequences Hive supports, but eventually decided against it and ended up leaving the behavior the same as it was before.

I could pretty easily cover the common cases if we want, or I suspect Hive uses `...hive.ql.parse.BaseSemanticAnalyzer.unescapeSQLString` or something similar so I could try using that.

Reply:
The first new commit rejects all escape sequences, and the second uses the function I mentioned above to translate all of them. I can squash them or drop the last one depending on what we want to do.